File: pawfinder-app-backend/pawfinder-shared/src/main/java/com/pawfinder/shared/domain/Conversation.java
package com.pawfinder.shared.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "conversations")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Conversation {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "alert_id", nullable = false)
    private Alert alert;

    @Column(name = "alert_id", insertable = false, updatable = false)
    private UUID alertId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id", nullable = false)
    private User owner;

    @Column(name = "owner_id", insertable = false, updatable = false)
    private UUID ownerId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "finder_id", nullable = false)
    private User finder;

    @Column(name = "finder_id", insertable = false, updatable = false)
    private UUID finderId;

    @Column(name = "sighting_id")
    private UUID sightingId;

    @Column(nullable = false)
    @Builder.Default
    private String status = "open";

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "closed_at")
    private LocalDateTime closedAt;

    @PrePersist
    protected void onCreate() {
        if (createdAt == null) {
            createdAt = LocalDateTime.now();
        }
    }
}

File: pawfinder_app_frontend/lib/domain/entities/conversation.dart
import 'package:equatable/equatable.dart';

/// Possible conversation statuses.
enum ConversationStatus { open, closed }

/// Domain entity representing a conversation between a pet owner and a finder.
class Conversation extends Equatable {
  final String id;
  final String alertId;
  final String ownerId;
  final String finderId;
  final ConversationStatus status;
  final DateTime createdAt;
  final DateTime? closedAt;

  const Conversation({
    required this.id,
    required this.alertId,
    required this.ownerId,
    required this.finderId,
    this.status = ConversationStatus.open,
    required this.createdAt,
    this.closedAt,
  });

  @override
  List<Object?> get props => [
        id,
        alertId,
        ownerId,
        finderId,
        status,
        createdAt,
        closedAt,
      ];
}
